.circleci/config.yml

```diff
@@ -23,23 +23,45 @@ jobs:
       - run:
           name: Install dependencies
           command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r .circleci/requirements.txt
+            sudo pip install -r .circleci/requirements.txt
 
       - save_cache:
-          paths:
-            - ./venv
           key: v1-dependencies-{{ checksum ".circleci/requirements.txt" }}
+          paths:
+            - ~/.cache 
 
       - run:
-          name: Linting
+          name: Run linter
           command: |
-            . venv/bin/activate
-            flake8 --max-line-length 160 --per-file-ignores='datascroller/__init__.py:F401' datascroller/
+            # stop the build if there are Python syntax errors or undefined names
+            flake8 datascroller --count --select=E901,E999,F821,F822,F823 \
+              --show-source --statistics
+            # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide      
+            flake8 datascroller --count --exit-zero --max-complexity=10 \
+              --max-line-length=127 --statistics
 
       - run:
-          name: Testing
+          name: Run Tests
           command: |
-            . venv/bin/activate
             pytest test/
+
+      - run:
+          name: init .pypirc
+          command: |
+            echo -e "[pypi]" >> ~/.pypirc
+            echo -e "username = $PYPI_USERNAME" >> ~/.pypirc
+            echo -e "password = $PYPI_PASSWORD" >> ~/.pypirc
+
+      - run:
+          name: create packages
+          command: |
+            python setup.py sdist
+            python setup.py bdist_wheel
+            
+      - deploy:
+          name: Upload to PyPi
+          command: |
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
```

Comment:
awesome!

Reply:
Yeah these [CircleCI environment variables](https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables) are sweet.